src/main/cpp/src/genomicsdb/genomicsdb_columnar_field.cc

```diff
@@ -0,0 +1,407 @@
+      break;
+    case VariantFieldTypeEnum::VARIANT_FIELD_CHAR:
+      m_print = GenomicsDBColumnarFieldPrintOperator<char, print_as_list>::print;
+      m_print_csv = GenomicsDBColumnarFieldPrintOperator<char, print_as_list>::print_csv;
+      break;
+    case VariantFieldTypeEnum::VARIANT_FIELD_STRING:
+      m_print = GenomicsDBColumnarFieldPrintOperator<char*, false>::print;
+      m_print_csv = GenomicsDBColumnarFieldPrintOperator<char*, print_as_list>::print_csv;
+      break;
+    default:
+      throw GenomicsDBColumnarFieldException(std::string("Unhandled type ")+m_element_type.name());
+  }
+}
+
+void GenomicsDBColumnarField::assign_function_pointers()
+{
+  auto variant_enum_type = VariantFieldTypeUtil::get_variant_field_type_enum_for_variant_field_type(m_element_type);
+  //Validity function
+  switch(variant_enum_type)
+  {
+    case VariantFieldTypeEnum::VARIANT_FIELD_INT:
+    case VariantFieldTypeEnum::VARIANT_FIELD_UNSIGNED:
+      m_check_tiledb_valid_element = GenomicsDBColumnarField::check_tiledb_valid_element<int>;
+      break;
+    case VariantFieldTypeEnum::VARIANT_FIELD_INT64_T:
+    case VariantFieldTypeEnum::VARIANT_FIELD_UINT64_T:
+      m_check_tiledb_valid_element = GenomicsDBColumnarField::check_tiledb_valid_element<int64_t>;
+      break;
+    case VariantFieldTypeEnum::VARIANT_FIELD_FLOAT:
+      m_check_tiledb_valid_element = GenomicsDBColumnarField::check_tiledb_valid_element<float>;
+      break;
+    case VariantFieldTypeEnum::VARIANT_FIELD_DOUBLE:
+      m_check_tiledb_valid_element = GenomicsDBColumnarField::check_tiledb_valid_element<double>;
+      break;
+    case VariantFieldTypeEnum::VARIANT_FIELD_BOOL:
+    case VariantFieldTypeEnum::VARIANT_FIELD_CHAR:
+    case VariantFieldTypeEnum::VARIANT_FIELD_STRING:
+      m_check_tiledb_valid_element = GenomicsDBColumnarField::check_tiledb_valid_element<char>;
+      break;
+    default:
+      throw GenomicsDBColumnarFieldException(std::string("Unhandled type ")+m_element_type.name());
+  }
+  //Singleton field
+  if(m_length_descriptor == BCF_VL_FIXED && m_fixed_length_field_num_elements == 1u)
+    assign_print_function_pointers<false>(variant_enum_type);
+  else
+    if(variant_enum_type == VariantFieldTypeEnum::VARIANT_FIELD_CHAR) //multi char field == string
+      assign_print_function_pointers<true>(VariantFieldTypeEnum::VARIANT_FIELD_STRING);
+    else
+      assign_print_function_pointers<true>(variant_enum_type);
+}
+
+void GenomicsDBColumnarField::move_buffer_to_live_list(GenomicsDBBuffer* buffer)
+{
+  assert(buffer && !(buffer->is_in_live_list()));
+  //Store neighbours in free list in tmp variables
+  auto next_in_free_list = buffer->get_next_buffer();
+  auto previous_in_free_list = buffer->get_previous_buffer();
+  //Insert into live list
+  //Since this will be the last element in the live list
+  buffer->set_next_buffer(0);
+  if(m_live_buffer_list_head_ptr == 0)
```

Comment:
A buffer should never go to the live list directly - always get into the free list, then moved